solidity/contracts/isms/hook/PolygonPosIsm.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+// ============ Internal Imports ============
+
+import {IInterchainSecurityModule} from "../../interfaces/IInterchainSecurityModule.sol";
+import {Message} from "../../libs/Message.sol";
+import {TypeCasts} from "../../libs/TypeCasts.sol";
+import {AbstractMessageIdAuthorizedIsm} from "./AbstractMessageIdAuthorizedIsm.sol";
+
+// ============ External Imports ============
+import {CrossChainEnabledPolygonChild} from "@openzeppelin/contracts/crosschain/polygon/CrossChainEnabledPolygonChild.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+/**
+ * @title PolygonPosIsm
+ * @notice Uses the native Polygon Pos Fx Portal Bridge to verify interchain messages.
+ */
+contract PolygonPosIsm is
+    CrossChainEnabledPolygonChild,
+    AbstractMessageIdAuthorizedIsm
+{
+    // ============ Constants ============
+
+    uint8 public constant moduleType =
+        uint8(IInterchainSecurityModule.Types.NULL);
+
+    // ============ Constructor ============
+
+    constructor(address _fxChild) CrossChainEnabledPolygonChild(_fxChild) {
+        require(
+            Address.isContract(_fxChild),
+            "PolygonPosIsm: invalid FxChild contract"
+        );
+    }
+
+    // ============ Internal function ============
+
+    uint256 public latestStateId;
+    address public latestRootMessageSender;
+    bytes public latestData;
+
+    /* function sendMessageToRoot(bytes memory message) public {
+        _sendMessageToRoot(message);
+    } */
```

Comment:
whats going on here?

Reply:
ah this is something I'd like to discuss, do i need to implement child to root message? (polygon pos to ethereum) which I don't think this should be implemented in ISM.